rules/rules_test.go

```diff
@@ -0,0 +1,44 @@
+package rules
+
+import (
+	"os"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	"github.com/googleapis/api-linter/testutil"
+)
+
+// This function maps testdata directories to their real directories. The convoluted mechanism for
+// doing so is an unfortunate consequence of maintaining compatibility when syncing internally.
+var testdatadir = func(lib string) string {
+	_, f, _, _ := runtime.Caller(0)
+	testdataDir := filepath.Dir(f) + string(os.PathSeparator) + "testdata" + string(os.PathSeparator)
```

Comment:
Why not return the `testdataDir` and they can use it as `filepath.Join(testdataDir, "api-common-protos")`? So we don't need to hard code "api-common-protos" or add more cases.

Reply:
my main concern is that in google3, the protos are spread out more, so it may not map cleanly to a single dir. for example, api-common-protos live in several directories spread out across google3/.

Also for this specific case, it wouldn't work, because the testdata dir would have to be `google3/` and they would then join `google3/api-common-protos` which doesn't exist

Reply:
Can we make including api-common-protos to the `proto_path` implicitly, similar to how we define `$PATH`? In other words, we always add `--proto_path=${testdataDir}/api-common-protos`, so:
1.  in the tests, we don't need to worry about where is `google/type/date.proto`;
2.  it allows us to do some future changes, e.g., we may want to rename `api-common-protos` or merge two folders;
3. if some tests need other proto files, (similar to `$PATH`), we append the path.

Reply:
I think what you're suggesting has some advantage in the form of convenience, but it's implicitly adding a dependency from testutil to api-common-protos. My argument against it is the same as the one against compiling the protos into the binary. I don't think we should couple these protos to our implementation.

There are very few tests that actually use those protos right now, so keeping it explicit is not costly at all (1 line of code per test that uses it). It would also be trivial to create a factory function in the `rules` package that creates a `FileDescriptorContext` with `AdditionalProtoPaths` pre-filled. Then all of the references to `api-common-protos` would be contained to the `rules` package (where the directory already lives).

By keeping the dependency explicit, we remove the expectation that anything added to `testdata` will automatically be available in tests, and tests that have different dependencies don't have to worry about any naming conflicts with the protos in `api-common-protos`

I guess if we move `api-common-protos` to `testutil/` it would make a bit more sense, but otherwise there's a strange cyclic dependency between testutil and rules. I'm thinking of `testutil` as a simple utility that allows you to specify your source code and its dependencies, rather than a fully packaged proto environment. If we decide to move the protos into testutil, I think we should make it an internal package. WDYT?

Reply:
I am confused -- I thought we were discussing about this `testdataDir` function -- I was suggesting something like 
```go
var common_proto_paths = func() []string {
 // register your proto paths here and we will return them
}
```
 to replace `testdataDir`.

Reply:
I can live with the current version, so I am resolving this conversation.